server/helpers/utilities.js

```diff
@@ -105,44 +107,82 @@ const configureCalendar = (data, startDate = null) => {
-    OVERTIME_NIGHT_START: 'Start',
-    OVERTIME_NIGHT_FINISH: 'Finish',
-  }[displayType])
-
-const displayedTasks = [
-  'DAY_START',
-  'DAY_FINISH',
-  'NIGHT_START',
-  'NIGHT_FINISH',
-  'OVERTIME_DAY_START',
-  'OVERTIME_DAY_FINISH',
-  'OVERTIME_NIGHT_START',
-  'OVERTIME_NIGHT_FINISH',
+    NIGHT_START: 'Night shift start',
+    NIGHT_FINISH: 'Night shift finish',
+    OVERTIME_DAY_START: 'Overtime start',
+    OVERTIME_DAY_FINISH: 'Overtime finish',
+    OVERTIME_NIGHT_START: 'Overtime night shift start',
+    OVERTIME_NIGHT_FINISH: 'Overtime night shift finish',
+  }[displayType]
+
+const fullDayActivities = [
+  // from cmd-api FullDayActivityType
+  { description: 'Rest Day', class: 'rest-day' },
+  { description: 'Annual Leave', class: 'absence' },
+  { description: 'Sick', class: 'illness' },
+  { description: 'Absence', class: 'absence' },
+  { description: 'Union Duties', class: 'tu-official' },
+  { description: 'Detached Duty', class: 'detached-duty' },
+  { description: 'TOIL', class: 'otherType' },
 ]
 
+const fullDayMatch = (desc) => {
+  const foundValue = fullDayActivities.find((a) => desc.startsWith(a.description))
+  return foundValue && foundValue.class
+}
+
+const unionFilter = (activity) =>
+  activity && (activity.startsWith('Union Duties') || activity.startsWith('Union Facility'))
+    ? 'Trade Union Official Duties'
+    : activity
+
 const processDay = (day) => {
-  const { date, details: rawTasks } = day
+  const { date, details, fullDayType } = day
   const dateMoment = moment(date)
   const today = dateMoment.isSame(moment(), 'day')
-  const format = 'hh:mm:ss'
-  const details = rawTasks.filter(
-    ({ displayType, start }) => displayedTasks.includes(displayType) && moment(start).format(format) !== '00:00:00',
-  )
-  // const sortedDetails = details.sort(d => d.start)
+
+  const isFullDay = fullDayType !== 'NONE' && fullDayType !== 'SHIFT' && fullDayType !== 'no-day'
+
+  let nightFinish = false
+
   details.forEach((detail) => {
-    const { displayType, displayTypeTime } = detail
-    const activity = `${getTaskText(displayType)} ${moment(displayTypeTime).format('HH:mm')}`
-    Object.assign(detail, { activity, displayType: displayType.toLowerCase() })
+    const { displayType, displayTypeTime, start, end, activity } = detail
+    const startText = start ? moment(start).format('HH:mm') : ''
+    const endText = end ? moment(end).format('HH:mm') : ''
+
+    if (displayType) {
+      const specialActivityColour = (!isFullDay && displayType.endsWith('START') && fullDayMatch(activity)) || ''
+      const showNightHr = nightFinish && displayType === 'NIGHT_START'
+      nightFinish = displayType === 'NIGHT_FINISH'
+      const durationColour = nightFinish
+        ? 'night_finish'
+        : (!isFullDay && displayType.endsWith('FINISH') && fullDayMatch(activity)) || ''
+      const activityText = `${getTaskText(displayType)}: ${moment(displayTypeTime).format('HH:mm')}`
+      Object.assign(detail, {
```

Comment:
not very nice to reassign `detail` each time like this.  Could you use `reduce` instead to end up with a new `details` object instead?  That way you wouldn't need a `nightFinish` variable either as you could pass the state through too...

Reply:
Just kept the existing approach, but yes, can do in the next iteration 👍